File: app/src/main/java/com/summer/itis/summerproject/ui/start/login/LoginActivity.java
package com.summer.itis.summerproject.ui.start.login;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.design.widget.TextInputLayout;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;


import com.google.firebase.auth.FirebaseAuth;
import com.summer.itis.summerproject.R;
import com.summer.itis.summerproject.ui.base.BaseActivity;
import com.summer.itis.summerproject.ui.member.member_item.PersonalActivity;
import com.summer.itis.summerproject.ui.start.registration.RegistrationActivity;

/**
 * Created by Ruslan on 18.02.2018.
 */

public class LoginActivity extends BaseActivity implements View.OnClickListener{

    private Button enterBtn;
    private TextView tvRegistration;
    private TextInputLayout tiUsername;
    private TextInputLayout tiPassword;
    private EditText etUsername;
    private EditText etPassword;

    private FirebaseAuth fireAuth;

    private LoginPresenter presenter;

    public static void start(@NonNull Context activity) {
        Intent intent = new Intent(activity, LoginActivity.class);
        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK).addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK);
        activity.startActivity(intent);
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        checkUserSession();


    }

    private void checkUserSession(){
        fireAuth = FirebaseAuth.getInstance();

        /*FirebaseUser user = fireAuth.getCurrentUser();
        if(user != null) {
            DatabaseReference reference = RepositoryProvider.getUserRepository().readUser(user.getUid());
            reference.addListenerForSingleValueEvent(new ValueEventListener() {
                @Override
                public void onDataChange(DataSnapshot dataSnapshot) {
                    User user = dataSnapshot.getValue(User.class);
                    NavigationPresenter.setCurrentUser(user);
                    goToBookList();
                }

                @Override
                public void onCancelled(DatabaseError databaseError) {

                }
            });
        } else {*/
            initViews();

    }

    private void initViews(){
        setContentView(R.layout.activity_login);

        enterBtn = findViewById(R.id.btn_enter);
        tvRegistration = findViewById(R.id.link_signup);

        enterBtn.setOnClickListener(this);
        tvRegistration.setOnClickListener(this);

        etUsername = findViewById(R.id.et_name);
        etPassword = findViewById(R.id.et_password);
        tiUsername = findViewById(R.id.ti_username);
        tiPassword =findViewById(R.id.ti_password);

        presenter = new LoginPresenter(this);
    }

    @Override
    public void onClick(View view) {

        switch(view.getId()){

            case R.id.btn_enter:
                String username = etUsername.getText().toString();
                String password = etPassword.getText().toString();
                presenter.signIn(username,password);
                break;

            case R.id.link_signup:
                goToRegistration();
                break;
        }
    }

    void goToProfile(){
        PersonalActivity.start(this);
    }

    private void goToRegistration() {
        RegistrationActivity.start(this);
    }

    public void showError() {
        tiUsername.setError(getString(R.string.enter_correct_name));
        tiPassword.setError(getString(R.string.enter_correct_password));
    }

    //get-set

    public FirebaseAuth getFireAuth() {
        return fireAuth;
    }

    public void setFireAuth(FirebaseAuth fireAuth) {
        this.fireAuth = fireAuth;
    }

    public TextInputLayout getTiUsername() {
        return tiUsername;
    }

    public void setTiUsername(TextInputLayout tiUsername) {
        this.tiUsername = tiUsername;
    }

    public TextInputLayout getTiPassword() {
        return tiPassword;
    }

    public void setTiPassword(TextInputLayout tiPassword) {
        this.tiPassword = tiPassword;
    }

    public EditText getEtUsername() {
        return etUsername;
    }

    public void setEtUsername(EditText etUsername) {
        this.etUsername = etUsername;
    }

    public EditText getEtPassword() {
        return etPassword;
    }

    public void setEtPassword(EditText etPassword) {
        this.etPassword = etPassword;
    }
}

File: app/src/main/java/com/summer/itis/summerproject/ui/start/login/LoginActivity.kt
package com.summer.itis.summerproject.ui.start.login


import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.design.widget.TextInputLayout
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError
import com.google.firebase.database.ValueEventListener
import com.summer.itis.summerproject.R
import com.summer.itis.summerproject.R.string.password
import com.summer.itis.summerproject.R.string.username
import com.summer.itis.summerproject.model.GameOne
import com.summer.itis.summerproject.repository.json.OldGameRepository
import com.summer.itis.summerproject.ui.base.BaseActivity
import com.summer.itis.summerproject.ui.member.member_item.PersonalActivity
import com.summer.itis.summerproject.ui.start.registration.RegistrationActivity
import com.summer.itis.summerproject.utils.Const.TAG_LOG
import java.util.*

/**
 * Created by Ruslan on 18.02.2018.
 */

class LoginActivity : BaseActivity(), View.OnClickListener {

    private var enterBtn: Button? = null
    private var tvRegistration: TextView? = null
    var tiUsername: TextInputLayout? = null
    var tiPassword: TextInputLayout? = null
    var etUsername: EditText? = null
    var etPassword: EditText? = null

    var fireAuth: FirebaseAuth? = null

    private var presenter: LoginPresenter? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        checkUserSession()


    }

    private fun checkUserSession() {
        fireAuth = FirebaseAuth.getInstance()

        initViews()

    }

    private fun initViews() {
        setContentView(R.layout.activity_login)

        enterBtn = findViewById(R.id.btn_enter)
        tvRegistration = findViewById(R.id.link_signup)

        enterBtn!!.setOnClickListener(this)
        tvRegistration!!.setOnClickListener(this)

        etUsername = findViewById(R.id.et_name)
        etPassword = findViewById(R.id.et_password)
        tiUsername = findViewById(R.id.ti_username)
        tiPassword = findViewById(R.id.ti_password)

        presenter = LoginPresenter(this)
//        enterBtn!!.performClick()
    }

    override fun onClick(view: View) {

        when (view.id) {

            R.id.btn_enter -> {
                val username = etUsername?.getText().toString();
                val password = etPassword?.getText().toString();
                /*val username = "rust@ma.ru"
                val password = "rustamka"*/
                presenter!!.signIn(username, password)
            }

            R.id.link_signup -> goToRegistration()
        }
    }

    internal fun goToProfile() {
        PersonalActivity.start(this)
    }

    private fun goToRegistration() {
        RegistrationActivity.start(this)
    }

    fun showError() {
        tiUsername!!.error = getString(R.string.enter_correct_name)
        tiPassword!!.error = getString(R.string.enter_correct_password)
    }

    companion object {

        fun start(activity: Context) {
            val intent = Intent(activity, LoginActivity::class.java)
            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK).addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK)
            activity.startActivity(intent)
        }
    }
}
